Don't set replacement if lhs is in the free symbols of the rhs

bobrenjc93 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #138915
* __->__ #139250

Fixes python test/dynamo/test_functions.py FunctionTests.test_is_integer

when we turn off specialize float on eager: https://github.com/pytorch/pytorch/pull/138915

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

bobrenjc93 committed:
Don't set replacement if lhs is in the free symbols of the rhs

[ghstack-poisoned]

ezyang reviewed (approved):
Also swing this one out. This one might be good to make a mini test case, it feels like it shouldn't be too difficult to do?

bobrenjc93 committed:
Update on "Don't set replacement if lhs is in the free symbols of the rhs"


Fixes python test/dynamo/test_functions.py FunctionTests.test_is_integer

when we turn off specialize float on eager: https://github.com/pytorch/pytorch/pull/138915

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

bobrenjc93 commented:
@pytorchbot merge
